usr.bin/truss/setup.c

```diff
@@ -76,37 +76,46 @@ static void	new_proc(struct trussinfo *, pid_t, lwpid_t);
 static struct procabi cloudabi32 = {
 	"CloudABI32",
 	SYSDECODE_ABI_CLOUDABI32,
+	sizeof(uint32_t),
 	STAILQ_HEAD_INITIALIZER(cloudabi32.extra_syscalls),
 	{ NULL }
 };
 
 static struct procabi cloudabi64 = {
 	"CloudABI64",
 	SYSDECODE_ABI_CLOUDABI64,
+	sizeof(uint64_t),
 	STAILQ_HEAD_INITIALIZER(cloudabi64.extra_syscalls),
 	{ NULL }
 };
 
 static struct procabi freebsd = {
 	"FreeBSD",
 	SYSDECODE_ABI_FREEBSD,
+	sizeof(void* __capability),
 	STAILQ_HEAD_INITIALIZER(freebsd.extra_syscalls),
 	{ NULL }
 };
 
-#ifdef __LP64__
+#if !defined(__SIZEOF_POINTER__)
+#error "Use a modern compiler."
+#endif
+
+#if __SIZEOF_POINTER__ > 4
 static struct procabi freebsd32 = {
 	"FreeBSD32",
 	SYSDECODE_ABI_FREEBSD32,
+	sizeof(uint32_t),
 	STAILQ_HEAD_INITIALIZER(freebsd32.extra_syscalls),
 	{ NULL }
 };
 #endif
 
-#ifdef __CHERI_PURE_CAPABILITY__
+#if __has_feature(capabilities)
```

Comment:
```suggestion
#if __SIZEOF_POINTER__ > 8
```
maybe? Then it's technically correct for a CHERI-64, which would have a freebsd32 but no freebsd64.

Reply:
I've added a _Static_assert instead, so that I can use `__has_feature(capabilities)` below for hybrid truss support without adding more ifdefs.